cli/src/cmd/cast/wallet/mod.rs

```diff
@@ -77,10 +87,39 @@ pub enum WalletSubcommands {
 impl WalletSubcommands {
     pub async fn run(self) -> eyre::Result<()> {
         match self {
-            WalletSubcommands::New { path, unsafe_password, .. } => {
+            WalletSubcommands::New { path, unsafe_password, mnemonic, word_count, .. } => {
                 let mut rng = thread_rng();
+                if mnemonic {
+                    let word_counts = [12, 15, 18, 21, 24];
+                    if !word_counts.contains(&word_count) {
```

Comment:
I don't think this is necessary to check ourselves as Mnemonic will just throw an error